.github/workflows/rust.yml

```diff
@@ -66,7 +66,7 @@ jobs:
       - name: cargo clippy
         if: matrix.toolchain == 'stable'
         run: cargo clippy ${{ matrix.profile }} ${{ matrix.features }} -- -D warnings
-      - name: cargo doc
+      - name: cargo doc --all-features
```

Comment:
This is just the name of the step, not the command that is to be run in the CI. The command to be run is below. What I recommend you doing is to keep the name as it was (as this same step is used when no features are enabled) and instead do the following:
1. Get rid of the `full` feature in `Cargo.toml`. Now that you have removed the `docs-rs` feature we can replace it with `--all-features` which is great!
2. In the CI file replace all instance of `--features full` to `--all-features` - this will also apply to the docs builds.

Also related to your change in `build.rs` I think it would be useful to prepend the command with `env DOCS_RS=1` so that we actually test the docs build in the environment `docs.rs` will use them.

Reply:
facepalm.

> Get rid of the full feature in Cargo.toml. Now that you have removed the docs-rs feature we can replace it with --all-features which is great!

Unless I miss something there is no way to use `all-features` for dependency https://doc.rust-lang.org/cargo/reference/features.html#dependency-features. So there is still a reason to have a "full" features. I will follow your choice.